nimble/host/src/ble_gap.c

```diff
@@ -3124,17 +3124,113 @@ ble_gap_adv_active(void)
 
 #if MYNEWT_VAL(BLE_EXT_ADV)
 static int
-ble_gap_ext_adv_params_tx(uint8_t instance,
```

Comment:
v1 and v2 structs have the same field order so we can have single helper to fill common fields

Reply:
So , since the structures become different, i made some changes in defination to club the common parts. Hope this is ok.